src/Debug/skeletonViewer.ts

```diff
@@ -743,11 +744,11 @@ export class SkeletonViewer {
     }
 
     private _buildLocalAxes(): void {
-        for (let axisMesh of this._localAxes) {
-            axisMesh.dispose();
+        if (this._localAxes) {
+            this._localAxes.dispose();
         }
-
-        this._localAxes = [];
+        

```

Comment:
Why removing the array of `LinesMesh` as it was before?

You will only dispose the latest mesh created, meaning all other meshes will still be displayed even if not needed anymore.

Reply:
You are right. That was a stupid mistake.

But now I started trying to make it into single mesh again. I got all the lines created and all, but I can't get the skinning to work. Is there any limitation stopping us from skinning a LinesMesh?

Reply:
It seems skinning is supported on a LineMesh, at least the vertex shader has the required bit of code.

Have you set the `skeleton` property of the LineMesh?

Reply:
This is my current code.

I've realized I need to apply some rotation to the points passed to `CreateLineSystem` as well, since it's not parented like before.

As you can see I do call `this._localAxes.skeleton = this.skeleton`.

```javascript
        let mwk: number[] = [];
        let mik: number[] = [];
        const vertsPerBone = 6;

        for (let i in this.skeleton.bones) {
            let bone = this.skeleton.bones[i];
            
            if (bone._index === -1 || (!this._boneIndices.has(bone.getIndex()) && !this.options.useAllBones)) {
                continue;
            }

            let boneAbsoluteRestTransform = new Matrix();
            let anchorPoint = new Vector3();

            this.getAbsoluteRestPose(bone, boneAbsoluteRestTransform);
            boneAbsoluteRestTransform.decompose(undefined, undefined, anchorPoint);

            let linePoints = [
                [anchorPoint, new Vector3(anchorPoint.x + size, anchorPoint.y, anchorPoint.z)],
                [anchorPoint, new Vector3(anchorPoint.x, anchorPoint.y + size, anchorPoint.z)],
                [anchorPoint, new Vector3(anchorPoint.x, anchorPoint.y, anchorPoint.z + size)]
            ];
            let lineColors = [[red, red], [green, green], [blue, blue]];

            lines.push(...linePoints);
            colors.push(...lineColors);
            
            for (let j = 0; j < vertsPerBone; j++) {
                mwk.push(1, 0, 0, 0);
                mik.push(bone.getIndex(), 0, 0, 0);
            }
        }

        this._localAxes = LinesBuilder.CreateLineSystem('localAxes', { lines: lines, colors: colors, updatable: true }, targetScene);
        this._localAxes.setVerticesData(VertexBuffer.MatricesWeightsKind, mwk, false);
        this._localAxes.setVerticesData(VertexBuffer.MatricesIndicesKind, mik, false);
        this._localAxes.skeleton = this.skeleton;
        this._localAxes.renderingGroupId = this.renderingGroupId;
        this._localAxes.parent = this.mesh;
        this._localAxes.computeBonesUsingShaders = this.options.computeBonesUsingShaders ?? true;
```

Reply:
Hard to tell...

Have you tried to get the rotation matrix (quaternion) from `boneAbsoluteRestTransform` instead?

Also, you say it does not work, but do you still see something moving? Meaning, does applying a skeleton actually work on the LineMesh?

Reply:
Yes, I tried using `boneAbsoluteRestTransform` but I didn't think it looked right. It accumulates rotations up the bone hierarchy, which I'm guessing is not desired in this case.

But perhaps I'm wrong. I guess it depends on how you interpret the local matrix passed to the bind pose (in the constructor that is). Is the rotation component of that matrix relative to the parent bone, or is it in world space?

Once this PR is working and correct, this can be made much more clear I think.

Reply:
> Yes, I tried using boneAbsoluteRestTransform but I didn't think it looked right. It accumulates rotations up the bone hierarchy, which I'm guessing is not desired in this case.

Yes, I think too.

So it would seem that `bone.getBindPose().getRotationMatrix();` is the right way to do it, don't know why it does not work...

Reply:
It does work, with your fix to hardware skinning. So thank you!